File: src/example-subscribe.py
#!venv/bin/python
import os, dotenv, autobahn.asyncio.component

DEFAULT_ENV = '.env.client';

if os.path.exists(DEFAULT_ENV):
    dotenv.load_dotenv(DEFAULT_ENV)

if (
    os.environ.get('WAMP_DEBUG') is None
    or os.environ.get('WAMP_HOST') is None
    or os.environ.get('WAMP_PORT') is None
    or os.environ.get('WAMP_AUTHID') is None
    or os.environ.get('WAMP_SECRET') is None
    or os.environ.get('WAMP_REALM') is None
): exit()

component = autobahn.asyncio.component.Component(
    transports='wss://'
    + os.environ.get('WAMP_HOST') + ':' + os.environ.get('WAMP_PORT'),
    realm=os.environ.get('WAMP_REALM'),
    authentication={
        'wampcra': {
            'authid': os.environ.get('WAMP_AUTHID'),
            'secret': os.environ.get('WAMP_SECRET'),
        }
    },
)


@component.subscribe(
    'test',
)
def test(*message):
    print('Received:', list(message))


if __name__ == '__main__':
    autobahn.asyncio.component.run(
        [component],
        True,
        'trace' if os.environ.get('WAMP_DEBUG') == 'true' else 'critical'
    )

File: src/example-subscribe.js
#!/usr/bin/node
const DEFAULT_ENV = '.env.client';

if (require('fs').existsSync(DEFAULT_ENV)) {
    require('dotenv').config({ path: DEFAULT_ENV })
}

if (
    !process.env.WAMP_DEBUG
    || !process.env.WAMP_HOST
    || !process.env.WAMP_PORT
    || !process.env.WAMP_AUTHID
    || !process.env.WAMP_SECRET
    || !process.env.WAMP_REALM
) {
    process.exit();
}

AUTOBAHN_DEBUG = (process.env.WAMP_DEBUG == "true");

var autobahn = require('autobahn');
var connection = new autobahn.Connection({
    url: 'wss://'+ process.env.WAMP_HOST +':'+ process.env.WAMP_PORT,
    realm: process.env.WAMP_REALM,
    authmethods: ['wampcra'],
    authid: process.env.WAMP_AUTHID,
    onchallenge: function (session, method, extra) {
        if (method === 'wampcra') {
            return autobahn.auth_cra.sign(
                process.env.WAMP_SECRET,
                extra.challenge
            );
        }
    }
});

connection.onopen = function(session) {

    session.subscribe('test', function(message) {
        console.log('Received:', message);
    });

};

connection.open();
